ukm-semantics/main/hooks/helpers.md

```diff
@@ -0,0 +1,26 @@
+```k
+
+module UKM-HOOKS-HELPERS-SYNTAX
+    syntax UkmInstruction ::= "ukmCancel"
+endmodule
+
+module UKM-HOOKS-HELPERS
+    imports private COMMON-K-CELL
+    imports private RUST-REPRESENTATION
+    imports private UKM-HOOKS-HELPERS-SYNTAX
+
+    syntax Identifier ::= "cancel_request"  [token]
+                        | "helpers"  [token]
+
+    rule
+        normalizedFunctionCall
+            ( :: helpers :: cancel_request :: .PathExprSegments
+            , .PtrList
+            )
+        => ukmCancel
+
+    rule (ukmCancel ~> _:KItem) => ukmCancel
```

Comment:
@virgil-serbanuta does this mean that, if we fail a require statement during the processing of a contract function, we remove everything from the K cell and end the execution just with `ukmCancel` in its place? Just to make sure, this is not supposed to revert anything that came before, just to prevent what is after it from being executed. Is that correct?

Reply:
This is an interesting question. And in EVM contracts we know that transaction failure in contract calls does not bubble up (the caller side will not revert) in some cases.
But in our case (The demo we are aiming for), we do not need to handle contract call at this stage.
We might be able to assume any failures should revert any changes made from the origin of this transaction call.
Does it make sense? (of course, we need to leave a `TODO: review` comment to reflect this decision for future reference).

Reply:
I am currently assuming that, in the normal ULM execution, the K cell contains only the current contract. This must be so because the K cell can contain only things that make sense in the current semantics, and calls between contracts must work the same way regardless of the semantics used by the contract.

Another reason for why it must be so is exactly what you're saying - you can't have a "bring your own language" framework if a contract can directly change the `<k>` cell of another contract. Or, similarly, if a stuck execution breaks everything.

I think that the framework is responsible for using the functions in the `UKM-SIGNATURE` module to extract execution state from the configuration in order to pass it to the caller contract, but, other than that, I think that two contracts have completely separate configurations.

This means, first, that it is safe to clear the `<k>` cell, since it will not be visible in the execution of other contracts. We can skip doing that, or we can do it in a different way (e.g. clear only up to a marker), but I think that the current implementation is fine. However, to make sure, I'll also ask a question in the shared rv-pi2 channel.

Reply:
Also, to explicitly answer @ACassimiro 's question: I am not removing things from the K cell in order to prevent execution (I could have done that by leaving the call to `fail` unevaluated, or something similar), I am removing things in order to make it easier to write tests.

Reply:
Thank you for answering, @virgil-serbanuta . We may need to wait for some positioning from the Pi2 project leads regarding whether we have full control over the K cell or must comply with any specific rules/behaviors related to it.

Reply:
Given the confirmation here: https://pi-squared-workspace.slack.com/archives/C07B7RED0GG/p1729165662372779 I will merge this as-is. If there are any objections, I will fix them in a follow-up PR.